reliability-engineering/terraform/modules/concourse-monitoring/grafana.tf

```diff
@@ -0,0 +1,173 @@
+    }
+  }
+
+  iam_instance_profile {
+    name = "${aws_iam_instance_profile.concourse_grafana.name}"
+  }
+
+  monitoring {
+    enabled = true
+  }
+
+  tag_specifications {
+    resource_type = "instance"
+
+    tags = {
+      Name       = "${var.deployment}-concourse-grafana"
+      Deployment = "${var.deployment}"
+    }
+  }
+
+  tag_specifications {
+    resource_type = "volume"
+
+    tags = {
+      Name       = "${var.deployment}-concourse-grafana"
+      Deployment = "${var.deployment}"
+    }
+  }
+
+  tags {
+    Deployment = "${var.deployment}"
+  }
+}
+
+resource "aws_autoscaling_group" "concourse_grafana" {
+  name = "${join("-", list(
+    "${aws_launch_template.concourse_grafana.id}",
+    "${aws_launch_template.concourse_grafana.latest_version}"
+  ))}"
+
+  max_size            = 2
+  min_size            = 0
+  desired_capacity    = 1
+  vpc_zone_identifier = ["${var.private_subnet_ids}"]
+
+  launch_template = {
+    id      = "${aws_launch_template.concourse_grafana.id}"
+    version = "$$Latest"
+  }
+
+  tag {
+    key                 = "Deployment"
+    value               = "${var.deployment}"
+    propagate_at_launch = true
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "random_string" "concourse_grafana_db_password" {
+  length  = 64
+  special = false
+}
+
+resource "aws_ssm_parameter" "concourse_grafana_db_password" {
+  name   = "/${var.deployment}/grafana/db-password"
+  type   = "SecureString"
+  key_id = "${aws_kms_key.concourse_grafana.key_id}"
+  value  = "${random_string.concourse_grafana_db_password.result}"
+}
+
+resource "random_string" "concourse_grafana_admin_password" {
+  length  = 32
+  special = false
+}
+
+resource "aws_ssm_parameter" "concourse_grafana_admin_password" {
+  name   = "/${var.deployment}/grafana/grafana-admin-password"
+  type   = "SecureString"
+  key_id = "${aws_kms_key.concourse_grafana.key_id}"
+  value  = "${random_string.concourse_grafana_admin_password.result}"
+}
+
+resource "aws_db_subnet_group" "concourse_grafana_db" {
+  name       = "${var.deployment}-concourse-grafana-db"
+  subnet_ids = ["${var.private_subnet_ids}"]
+}
+
+resource "aws_db_instance" "concourse_grafana_db" {
+  identifier                = "${var.deployment}-concourse-grafana"
+  allocated_storage         = 25
+  storage_type              = "gp2"
+  engine                    = "postgres"
+  instance_class            = "db.t2.small"
+  name                      = "grafana"
+  username                  = "grafana"
+  password                  = "${random_string.concourse_grafana_db_password.result}"
+  vpc_security_group_ids    = ["${aws_security_group.concourse_grafana_db.id}"]
+  db_subnet_group_name      = "${aws_db_subnet_group.concourse_grafana_db.name}"
+  final_snapshot_identifier = "${var.deployment}-concourse-grafana-final"
+  backup_retention_period   = 7
+  storage_encrypted         = true
+}
+
+data "template_file" "concourse_grafana_task_def" {
```

Comment:
really minor thing, but this isn't a task definition, it's a container definition